### PR TITLE
Add better pricing messaging on trial plans

### PR DESCRIPTION
**Pull request by jboland88**
## Proposed Changes

* This diff updates the language on the the Jetpack pricing page to better indicate how the $1 for the first month offers work.

## Testing Instructions

* Apply this patch to your testing environment or use the Calypso live link
* Check the `/pricing` page for the Jetpack Cloud environment
* Check the language next to the pricing for the VaultPress Backup and Social Basic, it should read something like "for the first month, then $10 /month billed yearly"
* Click the "More about" link for both these products to view the lightbox - the same language should be used within the lightbox

Screen Shot 2023-02-07 at 4 27 12 PM
Screen Shot 2023-02-07 at 4 26 58 PM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203923622825716

### CODE REVIEW

**Review by CodeyGuyDylan — CHANGES_REQUESTED** [review]
This was a problem that existed before, but is exacerbated with the even longer text. From 782px width all the way until 1118px width, the wrapping looks pretty bad
image
image

Other than this, the rest looks good. Do you think this is something to handle on this PR? Or leave that to another one?

**Comment by IanRamosC** [comment]
The messaging is more urgent, so I'd rather fix what @CodeyGuyDylan reported in a different PR.

**Review by IanRamosC — APPROVED** [review]

**Review by IanRamosC — APPROVED** [review]
Looks much better now. Thanks for the fix.

File: client/my-sites/plan-price/index.tsx
Comment (sirbrillig) [inline client/my-sites/plan-price/index.tsx]: 🤔 Not a huge deal, but is this supposed to be a React component, or a helper function? Typically React components are capitalized like this is, but the rest of it doesn't follow the rules of a functional component: it has multiple parameters instead of a single `props` parameter and it's being called inline at its call site rather than rendered with JSX.

I'd suggest picking one or the other.

Comment (jboland88) [reply client/my-sites/plan-price/index.tsx]: I intended it as a helper function - since this PR is fixing some pricing display text, we can flag this to clean up

File: client/components/jetpack/card/jetpack-product-card/display-price/time-frame.tsx
Comment (sirbrillig) [inline client/components/jetpack/card/jetpack-product-card/display-price/time-frame.tsx]: Since these new strings are not yet translated (I think?), you might want to use `hasTranslation` to keep the old string for the short term (unless you plan to wait for this PR until the translations complete before merging). Example: https://github.com/Automattic/wp-calypso/pull/63977/files

**Comment by IanRamosC** [comment]
@sirbrillig I've added a couple of other changes addressing your feedback

File: client/my-sites/plan-price/index.tsx
Comment (sirbrillig) [inline client/my-sites/plan-price/index.tsx:221]: I know this already merged but I realized something: this function doesn't need to be here; it's effectively a worse version of `formatCurrency()`, the main export of the `@automattic/format-currency` package, except it doesn't put the currency symbol in the right place. (Furthermore it doesn't make a lot of sense in this file since it's not related to `PlanPrice`.)

It would be good in a follow-up to remove this. Let me know if you'd like any assistance; I'm happy to help!

You can read more about the efforts to unify how prices are displayed here: pbOQVh-2WH-p2

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7799943

Thank you @jboland88 for including a screenshot in the description! This is really helpful for our translators.

**Comment by jornp** [comment]
👋 I found a small syntax issue with the `original_price` placeholder in the plural string — I've created a fix at https://github.com/Automattic/wp-calypso/pull/73079!

File: client/my-sites/plan-price/index.tsx
Comment (sirbrillig) [reply client/my-sites/plan-price/index.tsx:221]: Since it's been a while and I didn't want this to get lost (lest other folks start to use this function, not knowing about the better alternative) I've removed it in https://github.com/Automattic/wp-calypso/pull/73291

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7799943).

**Pull request merged**
